apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -17,33 +18,36 @@ defmodule Aecore.Chain.Worker do
 
   require Logger
 
-  @typep txs_index :: %{binary() => [{binary(), binary()}]}
+  @type txs_index :: %{binary() => [{binary(), binary()}]}
+  @max_refs 30 #upper limit for number of blocks is 2^max_refs
 
   def start_link(_args) do
     GenServer.start_link(__MODULE__, {}, name: __MODULE__)
   end
 
   def init(_) do
     genesis_block_hash = BlockValidation.block_header_hash(Block.genesis_block().header)
-    genesis_block_map = %{genesis_block_hash => Block.genesis_block()}
     genesis_chain_state = ChainState.calculate_and_validate_chain_state!(Block.genesis_block().txs, %{}, 0)
-    chain_states = %{genesis_block_hash => genesis_chain_state}
+    blocks_map = %{genesis_block_hash => 
```

Comment:
the name `blocks_map` is not fitting

Reply:
What do you suggest: "block_info_map", "block_data_map"?